### PR TITLE
[SPIRV] Break SPIRVVectorize pass into GenericVectorization and SPIRVVectorLowering passes.

### PR DESCRIPTION
**Pull request by qedawkins**
This removes vectorization patterns from SPIRVVectorize pass and delegate it to GenericVectorization.
It renames the old pass to SPIRVVectorLowering because it is mainly lowering vector ops.

This is a duplicate of https://github.com/openxla/iree/pull/14902 to push it through to completion.

### CODE REVIEW

**Review by MaheshRavishankar — COMMENTED** [review]
This mostly looks good to me. Ill let Lei or Jakub approve, but happy to stamp if you need me to.

**Comment by qedawkins** [comment]
This pass is fairly load bearing so I'll wait for their eyes. The hope is that it is pretty much NFC though.

File: compiler/src/iree/compiler/Codegen/SPIRV/Passes.td
Comment (kuhar) [inline compiler/src/iree/compiler/Codegen/SPIRV/Passes.td:117]: We should probably remove these constructors in the future and let tablegem autogenerate all of this code. But this is as a side note only, let's keep things as close to NFC as possible in this PR.

**Review by kuhar — APPROVED** [review]
LGTM

File: compiler/src/iree/compiler/Codegen/SPIRV/Passes.td
Comment (MaheshRavishankar) [reply compiler/src/iree/compiler/Codegen/SPIRV/Passes.td:117]: yes we should. I split up the TD file to be per backend to do that (someday). I

**Pull request merged**
